### DISCUSSION TITLE
Is OpenHaptics / 3DSystems Touch Supported?

### DISCUSSION THREAD

**Discussion by lsrodri** [PlusToolkit/PlusLib#945] [General]
Hi everyone,

Plus-2.8.0's release lists an OpenHaptics sensor. I tried connecting to it through both Plus Server Launcher and fCal, but the device wasn't listed. I tried adding this [configuration file](http://perk-software.cs.queensu.ca/plus/doc/nightly/dev/DeviceOpenHaptics.html), but I received the following error:

```
time|level|timeoffset|message|location
050522_183005.033|INFO|000.000000| System start timestamp: 889844| in E:\D\PSNP64b\PlusLib\src\PlusCommon\vtkPlusAccurateTimer.cxx(51)
050522_183008.911|ERROR|003.878000| Unknown device type: OpenHaptics. Supported devices: 3dConnexion, AuroraTracker, BkProFocusOem, ChRobotics, Epiphan, FakeTracker, GenericSerialDevice, ICCapturing, ImageProcessor, IntelRealSense, Microchip, MmfVideo, NDITracker, NoiseVideo, OpenCVVideo, OpenIGTLinkTracker, OpenIGTLinkVideo, OptiTrack, OpticalMarkerTracker, PhidgetSpatial, PolarisTracker, SavedDataSource, UsSimulator, VFWVideo, VirtualBufferedCapture, VirtualCapture, VirtualDiscCapture, VirtualMixer, VirtualSwitcher, VirtualTextRecognizer, VirtualVolumeReconstructor, WitMotionTracker| in E:\D\PSNP64b\PlusLib\src\PlusDataCollection\vtkPlusDeviceFactory.cxx(430)
050522_183008.924|ERROR|003.890000| Unable to create device: OpenHaptics| in E:\D\PSNP64b\PlusLib\src\PlusDataCollection\vtkPlusDataCollector.cxx(120)
050522_183008.924|ERROR|003.890000| No devices created. Please verify configuration file and any error produced.| in E:\D\PSNP64b\PlusLib\src\PlusDataCollection\vtkPlusDataCollector.cxx(134)
050522_183008.924|ERROR|003.890000| Unable to start collecting data!| in E:\D\PSNP64b\PlusApp\fCal\Toolboxes\QConfigurationToolbox.cxx(183)
```

As I understand it, if my device isn't supported, I should follow the [Add a device instructions](https://plustoolkit.github.io/devicecode), but since OpenHaptics is listed in the release, I would like to check if there is anything I might be missing. 

Thank you.

**Comment by adamrankin** [comment]
1) I don't think the OpenHaptics device interface is included in the standard release builds, so you would simply have to build your own Plus with it enabled to use it

2) The implementation of the device is a prototype at best, if it can do what you need, great! otherwise we welcome any improvements to the device interface

**Reply by lassoan** [reply]
FYI, we are experimenting with various ways of interacting with robotic devices in Slicer/SlicerIGT. During this summer, we'll evaluate connection via ROS, either using [ROS-IGTL bridge](https://github.com/openigtlink/ROS-IGTL-Bridge) or by creating a ROS node in a Slicer module (or maybe Plus).

@lucasiq what do you plan to do with the device?

**Reply by lsrodri** [reply]
@adamrankin Thank you, I am trying to build Plus at the moment. Is the current recommendation VS2017 with Qt 5.7 msvc2015_64 or should I use a newer version of Qt?

**Reply by lsrodri** [reply]
> FYI, we are experimenting with various ways of interacting with robotic devices in Slicer/SlicerIGT. During this summer, we'll evaluate connection via ROS, either using [ROS-IGTL bridge](https://github.com/openigtlink/ROS-IGTL-Bridge) or by creating a ROS node in a Slicer module (or maybe Plus).
> 
> @lucasiq what do you plan to do with the device?


@lassoan Thank you again for your guidance. If I may explain my context, my doctoral project is named "Visuo-Haptic Exploration of Biomedical Image Datasets", so I am hoping to build a Slicer module for direct haptic volume rendering. Could you tell me if you have similar goals in your development roadmap, or if your ROS implementation has other objectives? Thank you
